File: app/src/main/java/com/example/shieldnet/data/api/ShieldNetApi.java
package com.example.shieldnet.data.api;

import com.example.shieldnet.data.model.*;
import java.util.List;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface ShieldNetApi {

    @POST("auth/send-otp")
    Call<OtpSendResponse> sendOtp(@Body OtpRequest req);

    @POST("auth/verify-otp")
    Call<OtpVerifyResponse> verifyOtp(@Body OtpVerifyRequest req);

    @POST("workers/register")
    Call<RegisterResponse> registerWorker(@Body RegisterRequest req);

    @POST("risk/analyze")
    Call<FraudApiResponse> getRiskScore(@Body RiskRequest req);

    @GET("policies/active")
    Call<PolicyResponse> getActivePolicy(
            @Query("worker_id") String workerId
    );

    @POST("policies/create")
    Call<PolicyResponse> createPolicy(
            @Body PolicyCreateRequest req
    );

    @GET("claims/list")
    Call<List<ClaimResponse>> getClaims(
            @Query("worker_id") String workerId
    );

    @GET("status/triggers")
    Call<TriggerStatusResponse> getTriggerStatus(
            @Query("city") String city
    );
}

File: app/src/main/java/com/example/shieldnet/data/api/ShieldNetApi.kt
package com.example.shieldnet.data.api

import com.example.shieldnet.data.model.*
import retrofit2.Response
import retrofit2.http.*

interface ShieldNetApi {

    @POST("auth/send-otp")
    suspend fun sendOtp(@Body req: OtpRequest): Response<OtpSendResponse>

    @POST("auth/verify-otp")
    suspend fun verifyOtp(@Body req: OtpVerifyRequest): Response<OtpVerifyResponse>

    @POST("workers/register")
    suspend fun registerWorker(@Body req: RegisterRequest): Response<RegisterResponse>

    @POST("risk/analyze")
    suspend fun getRiskScore(@Body req: RiskRequest): Response<FraudApiResponse>


    @GET("policies/active")
    suspend fun getActivePolicy(
        @Query("worker_id") workerId: String
    ): Response<PolicyResponse>

    @POST("policies/create")
    suspend fun createPolicy(
        @Body req: PolicyCreateRequest
    ): Response<PolicyResponse>

    @GET("claims/list")
    suspend fun getClaims(
        @Query("worker_id") workerId: String
    ): Response<List<ClaimResponse>>

    @GET("status/triggers")
    suspend fun getTriggerStatus(
        @Query("city") city: String
    ): Response<TriggerStatusResponse>
}